pkgs/os-specific/linux/igh-ethercat-master/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib, stdenv, fetchFromGitLab, autoconf, automake, pkg-config, libtool, kmod, iproute2, kernel ? null }:
+
+stdenv.mkDerivation
+{
+  pname = "igh-ethercat-master";
+  version = "1.5-${kernel.modDirVersion}";
+
+  src = fetchFromGitLab {
+    owner = "etherlab.org";
+    repo = "ethercat";
+    rev = "4933150c17139e5e469197603984ca41ef9dbebe";
+    sha256 = "sha256-f60MWcbmOGe1Zflig9gxd52HVD8hyJ9AdopCUbsfecg=";
+  };
+
+  nativeBuildInputs = [ autoconf automake pkg-config libtool ] ++ kernel.moduleBuildDependencies;
+
+  makeFlags = [ "KDIR='${kernel.dev}/lib/modules/${kernel.modDirVersion}/build' all modules" ];
+
+  patches = [
+    ./ethercatctl.in.patch
+  ];
+
+  postPatch = ''
+    # Patch paths in ethercatctl script template
+    sed -i 's#LSMOD=/sbin/lsmod#LSMOD=${kmod}/bin/lsmod#' script/ethercatctl.in
+    sed -i 's#MODPROBE=/sbin/modprobe#LSMOD=${kmod}/bin/modprobe#' script/ethercatctl.in
+    sed -i 's#RMMOD=/sbin/rmmod#LSMOD=${kmod}/bin/rmmod#' script/ethercatctl.in
+    sed -i 's#MODINFO=/sbin/modinfo#LSMOD=${kmod}/bin/modinfo#' script/ethercatctl.in
+    sed -i 's#IP=/bin/ip#LSMOD=${iproute2}/bin/ip#' script/ethercatctl.in
+  '';
+
+  configurePhase = ''
+    bash ./bootstrap
+    ./configure --prefix=$out --with-linux-dir="${kernel.dev}/lib/modules/${kernel.modDirVersion}/build/" --disable-8139too --disable-e100 --disable-e1000 --disable-e1000e --disable-r8169 --disable-igb --enable-ccat
```

Comment:
Can we split this into preConfigure and configureFlags?

Reply:
I'd love to, but when I change it to this:

```
preConfigure = ''
  bash ./bootstrap
'';

configureFlags = [
  "--prefix=$out"
  "--with-linux-dir=\"${kernel.dev}/lib/modules/${kernel.modDirVersion}/build/\""
  "--disable-8139too"
  "--disable-e100"
  "--disable-e1000"
  "--disable-e1000e"
  "--disable-r8169"
  "--disable-igb"
  "--enable-ccat"
];
```

The build fails with error:

```
error: builder for '/nix/store/439fgy4j2mv50sbn5lrkff23v2wllzal-igh-ethercat-master-1.5.2-20221102-5.15.74.drv' failed with exit code 1;
       last 10 log lines:
       > checking if g++ PIC flag -fPIC -DPIC works... yes
       > checking if g++ static flag -static works... no
       > checking if g++ supports -c -o file.o... yes
       > checking if g++ supports -c -o file.o... (cached) yes
       > checking whether the g++ linker (ld) supports shared libraries... yes
       > checking dynamic linker characteristics... (cached) GNU/Linux ld.so
       > checking how to hardcode library paths into programs... immediate
       > checking whether to build kernel modules... yes
       > checking for Linux kernel sources...
       > configure: error: No configured Linux kernel sources in "/nix/store/hyqdfrp8m3287pq3xzip4yjrcb4wb3r2-linux-5.15.74-dev/lib/modules/5.15.74/build/"
       For full logs, run 'nix log /nix/store/439fgy4j2mv50sbn5lrkff23v2wllzal-igh-ethercat-master-1.5.2-20221102-5.15.74.drv'.
error: 1 dependencies of derivation '/nix/store/p6nk99sacbz0wmp4lach7h56acp8p117-kernel-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wajf27m0ll65w26gf1ci544l59dbiacd-linux-5.15.74-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0bhwlicclzj4fqgvs21y1gmnkbxrkwib-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fiwl835zdigk7xkblgpqnyyiqip1jk79-nixos-system-opennetic-22.11.20221107.dirty.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dczgx88gh1k2pnm697b0lnf0i29cpyif-ext4-fs.img.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gpkh82vkzhjfjqvv5wl9jqn4gabf47wp-nixos-sd-image-22.11.20221107.dirty-aarch64-linux.img.drv' failed to build
```

When I build it with `nix-build -A linuxKernel.packages.linux_5_15.igh-ethercat-master .`

Reply:
``--prefix`` should be added automtically. https://github.com/NixOS/nixpkgs/blob/db76c8377c352e25ee27846c06039ac9f7a0f341/pkgs/stdenv/generic/setup.sh#L1052


Also I am getting a hash mismatch for src and now the patch no longer applies.

and to fix the problem: The quoting is off. When using ``  "--with-linux-dir=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build/"`` it works which should be fine since the path practically cannot contain a space.

Reply:
Thanks, now it builds with preConfigure and configureFlags. Also fixed the patch hash.